pkg/processor/sourceprocessor/source_category_filler.go

```diff
@@ -76,21 +75,30 @@ func (f *sourceCategoryFiller) fill(attributes *pcommon.Map) {
 		return
 	}
 
-	valueTemplate := getAnnotationAttributeValue(f.annotationPrefix, sourceCategorySpecialAnnotation, attributes)
 	var templateAttributes []string
-	if valueTemplate != "" {
-		templateAttributes = extractTemplateAttributes(valueTemplate)
-	} else {
+	doesUseAnnotation := false
+
+	valueTemplate := getAnnotationAttributeValue(f.annotationPrefix, sourceCategorySpecialAnnotation, attributes)
+	if valueTemplate == "" {
 		valueTemplate = f.valueTemplate
-		templateAttributes = f.templateAttributes
+	} else {
+		doesUseAnnotation = true
 	}
 
 	prefix := getAnnotationAttributeValue(f.annotationPrefix, sourceCategoryPrefixAnnotation, attributes)
 	if prefix == "" {
 		prefix = f.prefix
+	} else {
+		doesUseAnnotation = true
 	}
 	valueTemplate = prefix + valueTemplate
 
+	if doesUseAnnotation {
```

Comment:
Why does it matter if we use annotations or not? Should prefix templating only work when using annotations?

Reply:
This is a good question to the author of the processor.
As I understand it: if we use the annotations, then we override the standard template value or the prefix, and then we can't use the preprocessed template attributes array.